lib/auth/index.ts

```diff
@@ -41,3 +41,17 @@ export {
 } from "./passwordReset";
 
 export { GET, POST, auth, signIn, signOut } from "./nextAuth";
+
+import { getAbility } from "@lib/privileges";
+
+// Creates a new custom Error Class
+export class AccessControlError extends Error {
+  public user: {
+    id: Promise<string>;
+  } = { id: getAbility().then((ability) => ability.userID) };
```

Comment:
From what I am seeing in the review process (so far), we could have simplified the `AccessControlError` object by passing the user identifier in the constructor. That would avoid having to deal with a promise.

Maybe there is one case (I did not see yet) that forced you to do it this way though.

Reply:
You're exactly right, my initial implementation did have the userID passed in as an arg to the constructor.  The issue was with the current implementation of AccessControlError across the app.  I'm hoping once the whole app is converted it'll be able to be simplified.